cmake/modules/extensions.cmake

```diff
@@ -4802,6 +4802,14 @@ endmacro()
 # loadable extensions (llexts).
 #
 
+# Usage:
+#   add_llext_target(<target_name>
+#                    OUTPUT  <output_file>
+#                    SOURCES <source_file>
+#                    [C_FLAGS <flags> ...]
+#                    [POSTPROC_CMD <cmd> [POSTPROC_ARGS <args> ...]]
```

Comment:
not fond of this, let's split the functions.

`add_llext_target()` creates the target and accepts source files, similar to CMake's add_custom_target / add_library.

But we should consider which one `add_llext_target()` resembles. With this PR I'm starting to think it probably more resembles add_library than custom_target.

Therefore I believe that instead of having `POSTPROC_CMD` then we should be have a function which resembles `add_custom_command(TARGET <target> PRE_/POST_BUILD)`.

As I see, we need finer granularity, so an idea could be:
```
add_llext_command(TARGET <llext_target> 
    [ PRE_BUILD | PRE_STRIP | POST_BUILD | <other> ] # Choose what is appropriate in this context
    COMMAND ....
)
```

That will also allow to gradually build the target and extra steps based on various other setting and not having to know everything from start.

Like:
```
add_llext_target(...)

if(<something>)
    add_llext_command(...)
endif()
```

Note: should we consider to rename `add_llext_target()` to `add_llext_library()` to make it more clear that we actually produce a lib.

Reply:
The product is an llext, not necessarily a library, it could be a partially linked elf representing a whole application to load, so I think library is a misrepresentation in this case. If shorter and closer to add _library is needed, then abbreviating to add_llext is fine, its a target to build

Reply:
> The product is an llext, not necessarily a library, it could be a partially linked elf representing a whole application to load, so I think adding the term library is a misrepresentation in this case.

and yet the term `lib` is used all over the place in this PR:
https://github.com/zephyrproject-rtos/zephyr/blob/37f60ed82af5c0ce18d0ad4c528724900b39f5d9/cmake/modules/extensions.cmake#L5172
https://github.com/zephyrproject-rtos/zephyr/blob/37f60ed82af5c0ce18d0ad4c528724900b39f5d9/cmake/modules/extensions.cmake#L5177
https://github.com/zephyrproject-rtos/zephyr/blob/37f60ed82af5c0ce18d0ad4c528724900b39f5d9/cmake/modules/extensions.cmake#L5195
etc.